src/main/kotlin/no/nav/amt/deltaker/bff/deltaker/DeltakerService.kt

```diff
@@ -0,0 +1,71 @@
+package no.nav.amt.deltaker.bff.deltaker
+
+import no.nav.amt.deltaker.bff.deltaker.api.DeltakerlisteDTO
+import no.nav.amt.deltaker.bff.deltaker.api.PameldingResponse
+import no.nav.amt.deltaker.bff.deltakerliste.DeltakerlisteMedArrangor
+import no.nav.amt.deltaker.bff.deltakerliste.DeltakerlisteRepository
+import org.slf4j.LoggerFactory
+import java.time.LocalDateTime
+import java.util.UUID
+
+class DeltakerService(
+    private val deltakerRepository: DeltakerRepository,
+    private val deltakerlisteRepository: DeltakerlisteRepository,
+) {
+    private val log = LoggerFactory.getLogger(javaClass)
+
+    fun opprettDeltaker(
+        deltakerlisteId: UUID,
+        personident: String,
+        opprettetAv: String,
+    ): PameldingResponse {
+        val deltakerliste = deltakerlisteRepository.getDeltakerlisteMedArrangor(deltakerlisteId) ?: throw NoSuchElementException("Fant ikke deltakerliste med id $deltakerlisteId")
+        val eksisterendeDeltaker = deltakerRepository.get(personident, deltakerlisteId)
+        if (eksisterendeDeltaker != null && !eksisterendeDeltaker.harSluttet()) {
+            log.warn("Deltakeren er allerede opprettet og deltar fortsatt")
+            return eksisterendeDeltaker.toPameldingResponse(deltakerliste)
+        }
+        val deltaker = getDeltaker(personident, deltakerlisteId, opprettetAv)
+        log.info("Oppretter deltaker med id ${deltaker.id}")
+        deltakerRepository.upsert(deltaker)
```

Comment:
Går det an å få denne til å returnere objektet som upsertes? Jeg har ikke helt klart å finne ut hvordan jeg gjør det med kotliquery :/

Reply:
Jeg tipper det er det samme som i `get`, så man bare legger på `returning *` i sqlen også, mapper man tilbake til deltaker som i `get`?

Reply:
Nå husker jeg hvorfor det ikke var så lett da jeg så på dette i går 🙈 Det er jo fordi vi gjør oppdateringer i to tabeller. Da må det kanskje bare være sånn..?